src/link.d

```diff
@@ -773,21 +808,40 @@ version (Windows)
         cmd = toWinPath(cmd);
         version (CRuntime_Microsoft)
         {
-            if (strchr(cmd, ' '))
+            // Open scope so dmd doesn't complain about alloca + exception handling
             {
-                // MSVCRT: spawn does not work with spaces in the executable
-                size_t cmdlen = strlen(cmd);
-                char* shortName = (new char[](cmdlen + 1)).ptr; // enough space
-                uint plen = GetShortPathNameA(cmd, shortName, cast(uint)cmdlen + 1);
-                if (plen > 0 && plen <= cmdlen)
-                    cmd = shortName;
+                // Use process spawning through the WinAPI to avoid issues with executearg0 and spawnlp
+                OutBuffer cmdbuf;
+                cmdbuf.writestring("\"");
+                cmdbuf.writestring(cmd);
+                cmdbuf.writestring("\" ");
+                cmdbuf.writestring(args);
+
+                STARTUPINFOA startInf;
```

Comment:
Its a shame that github hides comments of outdated diffs. As I said earlier, the documentation of STARTUPINFO and CreateProcess contradict each other. The documentation of STARTUPINFO states that if the dwFlags field of STARTUPINFO does not contain STARTF_USESTDHANDLES then the hStdInput, hStdOutput and hStdError fields of STARTUPINFO will be ignored and the default values will be used instead. And as I did run this code and it did not crash I assume that null are in fact correct values here. I will however do an additional test to see if redirected console output is correctly working when calling the linker in this way.